static/src/javascripts/lib/url.ts

```diff
@@ -0,0 +1,132 @@
+import memoize from 'lodash/memoize';
+import { hasPushStateSupport } from 'lib/detect';
+
+// TODO: typescript detect.js
+const supportsPushState = hasPushStateSupport() as boolean;
+
+/* commercial testing instrument */
+// Returns a map { <bidderName>: true } of bidders
+// according to the pbtest URL parameter
+
+const pbTestNameMap: () => Record<string, true | undefined> = memoize(
+	(): Record<string, undefined | true> =>
+		new URLSearchParams(window.location.search)
+			.getAll('pbtest')
+			.reduce<Record<string, undefined | true>>((acc, value) => {
```

Comment:
Can these records always be true here?

```suggestion
			.reduce<Record<string, true>>((acc, value) => {
```

Reply:
I started with this assumption, however, this is not quite right, because then TS thinks any key on the resulting object will be true.

This is obviously not the case, as some keys will return an `undefined` value.

Reply:
Ah gotcha!